macros/external/stage_external_sources.sql

```diff
@@ -71,6 +71,22 @@
 
 {% endmacro %}
 
+{% macro sqlserver__get_external_build_plan(source_node) %}
+
+    {% set build_plan = [] %}
+    
+    {%- set partitions = source_node.external.get('partitions', none) -%}
+    {% set create_or_replace = (var('ext_full_refresh', false)) %}
+    
+
+        {% set build_plan = [
+                dbt_external_tables.sqlserver__dropif(source_node),
+                dbt_external_tables.create_external_table(source_node)]%}
+    
+    {% do return(build_plan) %}
```

Comment:
Understood that SQLServer doesn't have a concept of "refreshing" external tables; neither does BigQuery. The way I've implemented this on other databases is such that `var('ext_full_refresh')` determines whether the operation is going to be _additive-only_ or _mutative_.

```sql
{% if create_or_replace %}
  {# drop all external tables and recreate / replace them #}
{% else %}
  {# create external tables if they don't already exist #}
  {# refresh external tables if they do, and if that's a thing this database does #}
{% endif %}
```

In practice, that may look like:
```sql
    {% set build_plan = [] %}
    
    {% set old_relation = adapter.get_relation(
        database = source_node.database,
        schema = source_node.schema,
        identifier = source_node.identifier
    ) %}
    
    {% set create_or_replace = (old_relation is none or var('ext_full_refresh', false)) %}

    {% if create_or_replace %}
        {% set build_plan = build_plan + [ 
            dbt_external_tables.dropif(source_node), 
            dbt_external_tables.create_external_table(source_node)
        ] %}
    {% else %}
        {% set build_plan = build_plan + dbt_external_tables.refresh_external_table(source_node) %}
    {% endif %}

    {% do return(build_plan) %}
```
Where:
```sql
{% macro sqlserver__refresh_external_table(source_node) %}
    {% do return([]) %}
{% endmacro %}
```

Reply:
maybe we can chat about this one...